Failed to serialize empty object in the Java client

**Describe the bug**

When we try serialize an empty object, we expect that receive something like that : 
`valueObject: {}
`
but we have `io.camunda.zeebe.client.api.command.InternalClientException`

**To Reproduce**

Add test to io.camunda.zeebe.client : 

```

  @Test
  public void toJson_shouldReturnEmptyObject() {
    final ZeebeObjectMapper zeebeObjectMapper = new ZeebeObjectMapper();

    final TestObject testObject = new TestObject();
    testObject.setObject(new Object());

    final String actual = zeebeObjectMapper.toJson(testObject);

    assertThat(actual).isEqualTo("{\"object\":{}}");
  }

  public final class TestObject {
    private Object object;

    public Object getObject() {
      return object;
    }

    public void setObject(final Object object) {
      this.object = object;
    }
  }

```

**Expected behavior**

"valueObject: {}"

**Log/Stacktrace**

<!-- If possible add the full stacktrace or Zeebe log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```

io.camunda.zeebe.client.api.command.InternalClientException: Failed to serialize object 'io.camunda.zeebe.client.ZeebeClientTest$TestObject@1c6e0a08' to json

	at io.camunda.zeebe.client.impl.ZeebeObjectMapper.toJson(ZeebeObjectMapper.java:91)
	at io.camunda.zeebe.client.ZeebeClientTest.toJson_shouldReturnEmptyObject(ZeebeClientTest.java:299)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:258)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:54)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class java.lang.Object and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) (through reference chain: io.camunda.zeebe.client.ZeebeClientTest$TestObject["object"])
	at com.fasterxml.jackson.databind.exc.InvalidDefinitionException.from(InvalidDefinitionException.java:77)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1300)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:46)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:29)
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:728)
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:774)
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:178)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	at com.fasterxml.jackson.databind.ObjectMapper._writeValueAndClose(ObjectMapper.java:4568)
	at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:3821)
	at io.camunda.zeebe.client.impl.ZeebeObjectMapper.toJson(ZeebeObjectMapper.java:88)
	... 33 more

```

</p>
</details>

**Environment:**
- OS: [e.g. Linux] 
- Zeebe Version: [8.2.0-SNAPSHOT] 
- https://github.com/camunda/zeebe/tree/main/clients

--- a/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeObjectMapper.java
+++ b/clients/java/src/main/java/io/camunda/zeebe/client/impl/ZeebeObjectMapper.java
@@ -19,6 +19,7 @@ import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 import io.camunda.zeebe.client.api.JsonMapper;
 import io.camunda.zeebe.client.api.command.InternalClientException;
 import java.io.IOException;
@@ -41,7 +42,9 @@ public final class ZeebeObjectMapper implements JsonMapper {
 
   public ZeebeObjectMapper(ObjectMapper objectMapper) {
     this.objectMapper = objectMapper;
-    this.objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    this.objectMapper
+        .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
   }
 
   @Override

--- a/clients/java/src/test/java/io/camunda/zeebe/client/impl/ZeebeObjectMapperTest.java
+++ b/clients/java/src/test/java/io/camunda/zeebe/client/impl/ZeebeObjectMapperTest.java
@@ -1,0 +1,48 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.impl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.camunda.zeebe.client.util.ClientTest;
+import org.junit.Test;
+
+public class ZeebeObjectMapperTest extends ClientTest {
+
+  @Test
+  public void shouldReturnEmptyObject() {
+    // Given
+    final ZeebeObjectMapper zeebeObjectMapper = new ZeebeObjectMapper();
+    final TestObject testObject = new TestObject();
+    testObject.setObject(new Object());
+    // When
+    final String actual = zeebeObjectMapper.toJson(testObject);
+    // Then
+    assertThat(actual).isEqualTo("{\"object\":{}}");
+  }
+
+  public static final class TestObject {
+    private Object object;
+
+    public Object getObject() {
+      return object;
+    }
+
+    public void setObject(final Object object) {
+      this.object = object;
+    }
+  }
+}

